src/node_jsrtapi.cc

```diff
@@ -364,69 +403,78 @@ napi_status napi_set_property(napi_env e, napi_value o,
   return napi_ok;
 }
 
-napi_status napi_define_property(napi_env e, napi_value o,
-    napi_property_descriptor* p) {
-  napi_value descriptor;
-  CHECK_NAPI(napi_create_object(e, &descriptor));
-
+napi_status napi_define_properties(napi_env e,
+                                   napi_value o,
+                                   int property_count,
+                                   const napi_property_descriptor* properties) {
   napi_propertyname configurableProperty;
   CHECK_NAPI(napi_property_name(e, "configurable", &configurableProperty));
```

Comment:
Since the calls to NAPI APIs here are all basically trivial wrappers around the JSRT APIs, I guess it's not saving much code if any. So I'd agree it would be better to rewrite this function using direct calls to JSRT.